From 229569a11b079df4c1a374a8f9ed3509c9ba2eb1 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:22 -0500
Subject: [wireless] Revert: mac80211: disable uAPSD if all ACs are under ACM

Message-id: <1418207389-7899-9-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101580
O-Subject: [RHEL7.1 PATCH 08/35] Revert "mac80211: disable uAPSD if all ACs are under ACM"
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Johannes Berg <johannes.berg@intel.com>

BZ#1155538

commit bb512ad0732232f1d2693bb68f31a76bed8f22ae upstream.

This reverts commit 24aa11ab8ae03292d38ec0dbd9bc2ac49fe8a6dd.

That commit was wrong since it uses data that hasn't even been set
up yet, but might be a hold-over from a previous connection.

Additionally, it seems like a driver-specific workaround that
shouldn't have been in mac80211 to start with.

Fixes: 24aa11ab8ae0 ("mac80211: disable uAPSD if all ACs are under ACM")
Reviewed-by: Luciano Coelho <luciano.coelho@intel.com>
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on Intel, Atheros and Ralink hardware.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 3345401..c8779f3 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -4355,8 +4355,7 @@ int ieee80211_mgd_assoc(struct ieee80211_sub_if_data *sdata,
  rcu_read_unlock();
 
  if (bss->wmm_used && bss->uapsd_supported &&
-     (sdata->local->hw.flags & IEEE80211_HW_SUPPORTS_UAPSD) &&
-     sdata->wmm_acm != 0xff) {
+     (sdata->local->hw.flags & IEEE80211_HW_SUPPORTS_UAPSD)) {
   assoc_data->uapsd = true;
   ifmgd->flags |= IEEE80211_STA_UAPSD_ENABLED;
  } else {
-- 
1.7.1